### PR TITLE
fix(demos):  on error partial fix

### PR DESCRIPTION
**Pull request by ShaMan123**
The page will not crash now, but we face a "already intialized canvas" error

### CODE REVIEW

File: src/components/CodeEditor/index.jsx [hunk 1]
```diff
@@ -57,29 +54,34 @@ export const CodeEditor = ({ code, children, canvasId }) => {
...
-  }, []);
+    const handler = () => console.log('TODO: handle disposing gracefully');
+    window.addEventListener('canvas_dispose', handler);
+    return () => window.removeEventListener('canvas_dispose', handler);
+  });
 
   return (
     <>
+      <Helmet>
+        <script type="module">{code}</script>
```
Comment (ShaMan123) [inline src/components/CodeEditor/index.jsx:83]: instead of eval

**Review by ShaMan123 — COMMENTED** [review]
done

**Comment by asturur** [comment]
image

after banging my head i deleted all the node_modules, killed package-lock file, reinstalled everything and not syntax highlighting works

File: src/components/CodeEditor/utils.js [hunk 2]
```diff
@@ -1,2 +1,2 @@
 export const functionToCodeString = (fn) =>
-  fn.toString().match(/function[^{]+\{([\s\S]*)\}$/)[1];
+  fn.toString().match(/function[^{]+\{\n([\s\S]*)\}$/)[1];
```
Comment (asturur) [inline src/components/CodeEditor/utils.js:2]: remove the empty new line at the start of the code editor

File: src/components/CodeEditor/index.jsx (hunk 1 above)
Comment (asturur) [reply src/components/CodeEditor/index.jsx:88]: not sure if there is a difference.
Is probably just slower?

Doesn't remove the problem of evaluating code written from the user.
Wraps the evaluation in 2-10 function wrappers.

**Pull request merged**